Code/Framework/AzFramework/AzFramework/Physics/Configuration/RigidBodyConfiguration.cpp

```diff
@@ -184,6 +193,35 @@ namespace AzPhysics
         return m_ccdEnabled;
     }
 
+    bool RigidBodyConfiguration::CcdReadOnly() const
+    {
+        if (auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get())
+        {
+            return !physicsSystem->GetDefaultSceneConfiguration().m_enableCcd || m_kinematic;
+        }
+        return m_kinematic;
+    }
+
+    AZStd::string RigidBodyConfiguration::GetCcdTooltip() const
+    {
+        if (m_kinematic)
+        {
+            return ccdDescription +
+                "<b>CCD cannot be enabled if the rigid body is kinematic, set the rigid body as non-kinematic to allow changes to be made.</b>";
+        }
+        return ccdDescription;
+    }
+
+    AZStd::string RigidBodyConfiguration::GetKinematicTooltip() const
+    {
+        if (m_ccdEnabled)
+        {
+            return kinematicDescription +
+                "<b>The rigid body cannot be set as Kinematic if CCD is enabled, disable CCD to allow changes to be made.</b>";
+        }
+        return kinematicDescription;
+    }
```

Comment:
This setup forces the system to allocate a new string every time the function is called. Since the outcomes of the function are known at compile time, it may be simpler to change this so that there is no memory allocation involved:

_(I'll write examples for just the kinematic part for clarity)_

1. Define all possible strings as `constexpr AZStd::string_view` constants:
```
    constexpr AZStd::string_view kinematicDescription =
        "When active, the rigid body is not affected by gravity or other forces and is moved by script. ";
    const AZStd::string kinematicDescriptionReadOnly =
        "When active, the rigid body is not affected by gravity or other forces and is moved by script. <b>The rigid body cannot be set as Kinematic if CCD is enabled, disable CCD to allow changes to be made.</b>"";
```

or even better, if the compiler lets you do it:

```
    constexpr AZStd::string_view kinematicDescription =
        "When active, the rigid body is not affected by gravity or other forces and is moved by script. ";
    const AZStd::string kinematicDescriptionReadOnly =
       kinematicDescription + "<b>The rigid body cannot be set as Kinematic if CCD is enabled, disable CCD to allow changes to be made.</b>"";
```

2. Change the function signature to return a `const AZStd::string_view`
```
const AZStd::string_view RigidBodyConfiguration::GetKinematicTooltip() const
```

3. Just return one string or the other in the function body
```
return (m_ccdEnabled) ? kinematicDescriptionReadOnly : kinematicDescription;
```

I'm writing this off the top of my mind, so the exact types may need to be adjusted a bit (maybe a string instead of string view, or it may be even simpler to just use `const char*`) but I think it conveys the base idea :)

Reply:
Thank you for this :) I changed it to a string view but no the compiler didn't let me concatenate string views :( 👎

Reply:
Makes sense now that I think about it again lol. One could replace it with a macro I guess, but given it's that simple I wouldn't overcomplicate it.